zenoh/src/net/runtime/adminspace.rs

```diff
@@ -715,7 +715,16 @@ fn routers_linkstate_data(context: &AdminContext, query: Query) {
     let tables = zread!(context.runtime.state.router.tables.tables);
 
     if let Err(e) = query
-        .reply(reply_key, tables.hat_code.info(&tables, WhatAmI::Router))
+        .reply(
+            reply_key,
+            context
```

Comment:
You have `context.runtime.state.router.tables` written above. A little factorisation could help to not clutter the code.